Git/commit.sublime-syntax

```diff
@@ -0,0 +1,84 @@
+%YAML 1.2
+---
+# Highlight regular git commits, merge commits, and tags.
+
+name: Git Commit
+file_extensions:
+  - COMMIT_EDITMSG
+  - MERGE_MSG
+  - TAG_EDITMSG
+  - syntax_test_git_commit
+  - syntax_test_git_merge
+  - syntax_test_git_tag
+scope: source.gitcommit
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: commit-message
+
+  branch-line:
+    - match: \bOn branch\s+(.*)
+      captures:
+        1: support.type.branch-name.gitcommit
+
+  change:
+    - match: (new file|modified|deleted)\s*(:)\s*(.*)
+      captures:
+        1: keyword.change-list.change.gitcommit
+        2: punctuation.separator.change.gitcommit
+        3: keyword.change-list.file.gitcommit
+
+  change-list:
+    - include: change-list-header
+    - include: change
+
+  change-list-header:
+    - match: '\bChanges to be committed\s*(:)'
+      scope: keyword.change-list.heading.gitcommit
+      captures:
+        1: punctuation.separator.change-list.gitcommit
+
+  comments:
+    - match: '#'
+      scope: punctuation.definition.comment.gitcommit
+      push:
+        - meta_scope: comment.line.gitcommit
+        - match: \n
+          pop: true
+        - include: change-list
+        - include: branch-line
+        - include: date-line
+
+  commit-message:
+    # this absurd regex is necessary to:
+    #   1. match comments in the subject line
+    #   2. only style the first 50 chars in the subject line
+    - match: ([^#\n]{,50})[^#\n]*((#).*)?
```

Comment:
I think you can leave out the newline characters here.

Reply:
Actually, I can't remove them. My (limited) understanding is that without the newlines, the regex is matching up to _and including_ the newline char. This prevents the first pushed context (`\n`) from matching and highlighting non-empty lines as invalid.
#### With
![working](https://user-images.githubusercontent.com/1250684/30006670-622a4c14-90b2-11e7-82b9-d49852086842.png)


#### Without
![broken](https://user-images.githubusercontent.com/1250684/30006656-2e31fef2-90b2-11e7-9bb5-934badda059b.png)

Reply:
Good point.